### PR TITLE
ceph-extra-exporter: improve README.md

### PR DESCRIPTION
**Pull request by llamerada-jp**

### CODE REVIEW

File: ceph-extra-exporter/README.md [hunk 1]
```diff
@@ -3,6 +3,33 @@ ceph-extra-exporter
...
 
+## How to work
+
+ceph-extra-exporter get metrics using `ceph`, `rados-gw-admin`, and related commands and export metric values like as follows.
+
+- Run `ceph` commands etc written in the code.
+- Modify output JSON to specify format using jq filter.
+- Decode formatted JSON and export these values as the metrics.
+
+The commands to get values and arguments for jq are hard-coded, so to add metrics, please modify the program by reading [`How to add metrics`](#how-to-add-a-metrics) section.
```
Comment (peng225) [inline ceph-extra-exporter/README.md:14]: Though it is not a problem of this PR, "how-to-add-a-metrics" is a broken English.
It should be "how-to-add-metrics" or "how-to-add-a-metric", and the name of the related section should be also changed.

File: ceph-extra-exporter/README.md [hunk 2]
```diff
@@ -3,6 +3,33 @@ ceph-extra-exporter
...
+
+| Option | Type | Default value | Description                   |
+| ------ | ---- | ------------- | ----------------------------- |
+| `port` | uint | `8080`        | port number to export metrics |
+
+API endpoints are:
+
+| Path        | Description                  |
+| ----------- | ---------------------------- |
+| /v1/health  | This path for liveness probe |
```
Comment (peng225) [inline ceph-extra-exporter/README.md:30]:
```suggestion
| /v1/health  | The path for liveness probe |
```

File: ceph-extra-exporter/README.md [hunk 3]
```diff
@@ -3,6 +3,33 @@ ceph-extra-exporter
 
 `ceph-extra-exporter` exposes extra metrics using some Ceph commands.
 
+## How to work
+
+ceph-extra-exporter get metrics using `ceph`, `rados-gw-admin`, and related commands and export metric values like as follows.
```
Comment (peng225) [inline ceph-extra-exporter/README.md:8]:
```suggestion
ceph-extra-exporter gets metrics using `ceph`, `rados-gw-admin`, and related commands and export metric values like as follows.
```

File: ceph-extra-exporter/README.md [hunk 4]
```diff
@@ -3,6 +3,33 @@ ceph-extra-exporter
...
+
+- Run `ceph` commands etc written in the code.
```
Comment (peng225) [inline ceph-extra-exporter/README.md:10]:
```suggestion
- Run commands such as `ceph` written in the code. The output is in JSON format.
```

File: ceph-extra-exporter/README.md [hunk 5]
```diff
@@ -3,6 +3,33 @@ ceph-extra-exporter
...
+- Modify output JSON to specify format using jq filter.
```
Comment (peng225) [inline ceph-extra-exporter/README.md:11]: I could not understand the meaning of this explanation.
I would be happy if you could tell me the specific lines of Go code that this explanation refers to.

Comment (peng225) [reply ceph-extra-exporter/README.md:11]: @llamerada-jp Could you give me a reply to this comment?

File: ceph-extra-exporter/README.md [hunk 6]
```diff
@@ -3,6 +3,34 @@ ceph-extra-exporter
...
+- Run commands such as `ceph` written in the code. The output is in JSON format.
+- Modify output JSON to specify format using jq filter.
+- Decode formatted JSON and export these values as the metrics.
+
+This logic is implemented [here](https://github.com/cybozu/neco-containers/blob/91eb3f4771cb0e02513e078e60c98358a88bd280/ceph-extra-exporter/executer.go#L87-L126).
+The commands to get values and arguments for jq are [hard-coded](https://github.com/cybozu/neco-containers/blob/91eb3f4771cb0e02513e078e60c98358a88bd280/ceph-extra-exporter/main.go#L18-L54), so to add metrics, please modify the program by reading [`How to add metrics`](#how-to-add-metrics) section.
+
+## How to run
+
+To use `ceph` and `radosgw-admin` commands in the docker container of ceph-extra-exporter, configuration files for ceph commands are required. To retrieve those files, deploy ceph-extra-exporter container as a sidecar of rook toolbox sharing the `/etc/ceph` directory.
```
Comment (daichimukai) [inline ceph-extra-exporter/README.md:19]: > To retrieve those files, deploy ceph-extra-exporter container as a sidecar of rook toolbox sharing the `/etc/ceph` directory.

This sentence is too concrete. The ceph-extra-exporter requires `ceph.conf` as you noted, but it need not be a sidecar of rook toolbox.

File: ceph-extra-exporter/README.md [hunk 7]
```diff
@@ -3,6 +3,34 @@ ceph-extra-exporter
...
+
+Command-line options are:
+
+| Option | Type | Default value | Description                   |
+| ------ | ---- | ------------- | ----------------------------- |
+| `port` | uint | `8080`        | port number to export metrics |
```
Comment (daichimukai) [inline ceph-extra-exporter/README.md:25]: We cannot pass anything other than "string" to executables...

File: ceph-extra-exporter/README.md [hunk 8]
```diff
@@ -3,6 +3,34 @@ ceph-extra-exporter
...
+
+API endpoints are:
+
+| Path        | Description                 |
+| ----------- | --------------------------- |
+| /v1/health  | The path for liveness probe |
+| /v1/metrics | Exporting metrics           |
```
Comment (daichimukai) [inline ceph-extra-exporter/README.md:32]:
```suggestion
| /v1/health  | the path for liveness probe |
| /v1/metrics | exporting metrics           |
```

Just for consistency.

File: ceph-extra-exporter/README.md (hunk 5 above)
Comment (peng225) [reply ceph-extra-exporter/README.md:11]: @llamerada-jp Sorry, but I just wanted to see the code to understand this explanation. I think you need not to write the Go source info in the document itself.

Anyway, now I believe I understand the meaning. How about changing this line as follows?

```suggestion
- Modify the output JSON to the specified format using jq filter.
```

**Review by peng225 — COMMENTED** [review]
See [this comment](https://github.com/cybozu/neco-containers/pull/979#discussion_r1113700256).

**Review by peng225 — DISMISSED** [review]

File: ceph-extra-exporter/README.md [hunk 9]
```diff
@@ -3,6 +3,33 @@ ceph-extra-exporter
...
 
+## How to work
+
+ceph-extra-exporter gets metrics using `ceph`, `rados-gw-admin`, and related commands and export metric values like as follows.
+
+- Run commands such as `ceph` written in the code. The output is in JSON format.
+- The above command outputs are inherent JSON for each of them. So convert it to specified structure JSON using jq filter.
+- Decode formatted JSON and export these values as the metrics.
+
+The commands to get values and arguments for jq are hard-coded, so to add metrics, please modify the program by reading [`How to add metrics`](#how-to-add-metrics) section.
```
Comment (daichimukai) [inline ceph-extra-exporter/README.md:14]:
```suggestion
The commands to get values and arguments for jq are hard-coded. To add metrics, please modify the program by reading [`How to add metrics`](#how-to-add-metrics) section.
```

File: ceph-extra-exporter/README.md [hunk 10]
```diff
@@ -3,6 +3,33 @@ ceph-extra-exporter
...
+
+## How to run
+
+To use `ceph` and `radosgw-admin` commands in the docker container of ceph-extra-exporter, configuration file (usually it is in `/etc/ceph/ceph.conf`) is required. An example of how to setup the file, deploy rook toolbox as a sidecar container of ceph-extra-exporter and sharing the `/etc/ceph` directory.
```
Comment (daichimukai) [inline ceph-extra-exporter/README.md:18]:
```suggestion
To use `ceph` and `radosgw-admin` commands in the docker container of ceph-extra-exporter, configuration file (usually it is in `/etc/ceph/ceph.conf`) is required. For example, you can deploy a rook toolbox as a sidecar container of ceph-extra-exporter and share the `/etc/ceph` directory between them.
```

**Review by peng225 — DISMISSED** [review]

**Review by daichimukai — DISMISSED** [review]

**Comment by daichimukai** [comment]
This PR contains only changes in README.md so literally *all* CI jobs are skipped? Then merging is blocked forever. lol

**Review by daichimukai — APPROVED** [review]

**Pull request merged**